File: backend/src/main/kotlin/com/example/backend/models/History.kt
package com.example.backend.models

import jakarta.persistence.*
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "history")
class History {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "produit_id")
  var produit: com.example.backend.models.Produit? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  var user: com.example.backend.models.User? = null

  @Column(name = "date_histo")
  var dateHisto: LocalDateTime? = null

  @Column(name = "description", length = 64)
  var description: String? = null

  @Column(name = "quantite", nullable = false)
  var quantite: Int? = null

  @Column(name = "typeHisto", nullable = false, length = 64)
  var typeHisto: String? = null
}

File: backend_py/pharmaPython/app/models/history.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class History(Base):
  __tablename__ = "history"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  produit_id = Column(Integer, ForeignKey("produit.id"))
  produit = relationship("Produit", lazy="select")
  user_id = Column(Integer, ForeignKey("user.id"))
  user = relationship("User", lazy="select")
  date_histo = Column(DateTime)
  description = Column(String(64))
  quantite = Column(Integer, nullable=False)
  type_histo = Column(String(64), nullable=False)
